koku/api/report/ocp/capacity/cluster_capacity.py

```diff
@@ -83,28 +85,117 @@ def capacity_annotations(self):
     def count_units(self):
         return self.report_type_map.get("count_units_key")
 
+    @property
+    def capacity_count_key(self):
+        return self.report_type_map.get("capacity_count_key")
+
     @property
     def count_annotations(self):
         return self.capacity_aggregate.get("cluster_instance_counts", {})
 
     def __post_init__(self):
         self._populate_count_values()
 
+    def _aggregate_capacity_count_by_cluster(self, node_instance_counts):
+        """
+        Helper function to aggregate capacity count by cluster.
+
+        Aggregates the maximum capacity count for each node within a cluster from a given set of node instances.
+
+        Args:
+            node_instance_counts (QuerySet): Django queryset containing annotated node instance counts
+                                                with cluster, node, usage_start and capacity_count fields.
+
+        Side effects:
+            - Modifies `self.count_by_cluster` to include the total capacity count for each cluster keyed by cluster.
+
+        """
+
+        unique_cluster_node_counts = {}
+        for count in node_instance_counts:
+            cluster, node, capacity_count = (count["cluster"], count["node"], count["capacity_count"])
+
+            # count each node per cluster once
+            # use max capacity count if a node has multiple values
+            if cluster not in unique_cluster_node_counts:
+                unique_cluster_node_counts[cluster] = {node: capacity_count}
+            if node not in unique_cluster_node_counts[cluster]:
+                unique_cluster_node_counts[cluster][node] = capacity_count
+            else:
+                unique_cluster_node_counts[cluster][node] = max(
+                    capacity_count, unique_cluster_node_counts[cluster].get(node, 0)
+                )
```

Comment:
I believe you can also eliminate a bunch of branches here by using a defaultdict for the `unique_cluster_node_counts`:
```suggestion
        unique_cluster_node_counts = defaultdict(lambda: defaultdict(Decimal))
        for count in node_instance_counts:
            cluster, node, capacity_count = (count["cluster"], count["node"], count["capacity_count"])

            # count each node per cluster once
            # use max capacity count if a node has multiple values
            unique_cluster_node_counts[cluster][node] = max(
                capacity_count, unique_cluster_node_counts[cluster].get(node, 0)
            )
```